stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -107,31 +155,42 @@ import SwiftShims
 // dictionary buffer are bounds-checked, this scheme never compromises memory
 // safety.
 //
+//
 // Bridging
 // ========
 //
 // Bridging `NSDictionary` to `Dictionary`
 // ---------------------------------------
 //
+// FIXME(eager-bridging): rewrite this based on modern constraints.
+//
 // `NSDictionary` bridges to `Dictionary<NSObject, AnyObject>` in `O(1)`,
 // without memory allocation.
 //
 // Bridging `Dictionary` to `NSDictionary`
 // ---------------------------------------
 //
-// `Dictionary<K, V>` bridges to `NSDictionary` iff both `K` and `V` are
-// bridged.  Otherwise, a runtime error is raised. Because 
-// _NativeDictionaryStorage inherits from ManagedBuffer, it can't also
-// inherit from NSDictionary (_NSDictionaryCore). To have a value that does,
-// we wrap the _NativeDictionaryStorage in another class, 
-// _NativeDictionaryBridgingStorage, which incurs an allocation, but is O(1).
+// `Dictionary<K, V>` bridges to `NSDictionary` in O(1)
+// but may incur an allocation depending on the following conditions:
+//
+// * If the Dictionary is freshly allocated without any elements, then it
```

Comment:
Is there a difference between “freshly allocated without any elements” and “empty?”  If not, please say ”empty.”  If so, please  be clear about what the distinction is.